code/controllers/CMSMain.php

```diff
@@ -78,22 +78,21 @@ public function init() {
 
 		Requirements::css(CMS_DIR . '/css/screen.css');
 		Requirements::customCSS($this->generatePageIconsCss());
-
-		Requirements::combine_files(
-			'cmsmain.js',
-			array_merge(
-				array(
-					CMS_DIR . '/javascript/CMSMain.js',
-					CMS_DIR . '/javascript/CMSMain.EditForm.js',
-					CMS_DIR . '/javascript/CMSMain.AddForm.js',
-					CMS_DIR . '/javascript/CMSPageHistoryController.js',
-					CMS_DIR . '/javascript/CMSMain.Tree.js',
-					CMS_DIR . '/javascript/SilverStripeNavigator.js',
-					CMS_DIR . '/javascript/SiteTreeURLSegmentField.js'
-				),
-				Requirements::add_i18n_javascript(CMS_DIR . '/javascript/lang', true, true)
-			)
-		);
+		Requirements::add_i18n_javascript(CMS_DIR . '/javascript/lang', true, true);
+		Requirements::javascript(CMS_DIR . '/javascript/dist/bundle-lib.js', [
+			'provides' => [
+				CMS_DIR . '/javascript/dist/AssetAdmin.DragDrop.js',
+				CMS_DIR . '/javascript/dist/AssetAdmin.js',
```

Comment:
AssetAdmin* files are specific to AssetAdmin, so they probably shouldn't be in the pages section.